### PR TITLE
provider/aws: Add validation for aws_elb.name

### PR DESCRIPTION
**Pull request by radeksimko**
http://docs.aws.amazon.com/cli/latest/reference/elb/create-load-balancer.html

screen shot 2015-06-26 at 15 04 56

### CODE REVIEW

**Comment by phinze** [comment]
> must be unique within your AWS account

Someday maybe we can actually do an API query to verify this. Someday. Maybe. :grinning:

**Comment by phinze** [comment]
LGTM

**Comment by radeksimko** [comment]
> Someday maybe we can actually do an API query to verify this. Someday. Maybe.

I see your point, but the beauty of these validations is that it can all be done offline :) so to keep that benefit, we'd have to save all existing ELBs in the account somewhere... :-/

**Comment by phinze** [comment]
Good point. Something to think on.

My feeling is that it seems like we should do everything we can to catch as many potential errors in the Plan step instead of the Apply step - that would be the difference in doing a pre-Describe earlier. Of course there's always going to be a race condition. We can discuss more once we get to that layer of niceties. Someday. :+1: :grinning:

File: builtin/providers/aws/resource_aws_elb.go [hunk 1]
```diff
@@ -25,6 +26,26 @@ func resourceAwsElb() *schema.Resource {
 				Type:     schema.TypeString,
 				Required: true,
 				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if !regexp.MustCompile(`^[0-9a-z-]$`).MatchString(value) {
```
Comment (radeksimko) [inline builtin/providers/aws/resource_aws_elb.go]: @catsby I overlooked this one too... :/

Comment (catsby) [inline builtin/providers/aws/resource_aws_elb.go]: no worries, they're getting sorted out :smile:

**Comment by pmoust** [comment]
woops, we 've got about 12 ELBs that are `[0-9A-Za-z-]`.
@radeksimko: Any reason why you went for `[0-9a-z-]`?
I went in to test v0.6.0 and got this little surprise.

**Comment by blairham** [comment]
I have the same question as @pmoust.

**Pull request merged**